theatre/core/src/projects/DefaultAssetStorage.ts

```diff
@@ -0,0 +1,172 @@
+import * as idb from 'idb-keyval'
+import type Project from './Project'
+import {val} from '@theatre/dataverse'
+import type {IAssetStorageConfig} from './Project'
+// @ts-ignore
+import blobCompare from 'blob-compare'
+
+function getAllPossibleAssetIDs(project: Project) {
+  const sheets = Object.values(val(project.pointers.historic.sheetsById))
+  const staticValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.staticOverrides.byObject ?? {}))
+    .flatMap((overrides) => Object.values(overrides ?? {}))
+  const keyframeValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.sequence?.tracksByObject ?? {}))
+    .flatMap((tracks) => Object.values(tracks?.trackData ?? {}))
+    .flatMap((track) => track?.keyframes)
+    .map((keyframe) => keyframe?.value)
+
+  const allValues = [...staticValues, ...keyframeValues].filter(
+    // value is string, and is unique
+    (value, index, self) =>
+      typeof value === 'string' && self.indexOf(value) === index,
+  ) as string[]
+
+  return allValues
+}
+
+export const createDefaultAssetStorageConfig = ({
+  project,
+  baseUrl = '',
+}: {
+  project: Project
+  baseUrl?: string
+}): IAssetStorageConfig => {
+  return {
+    coreAssetStorage: {
+      getAssetUrl: (assetId: string) => `${baseUrl}/${assetId}`,
+    },
+    createStudioAssetStorage: async () => {
+      // Check for support.
+      if (!('indexedDB' in window)) {
+        console.log("This browser doesn't support IndexedDB.")
+
+        return {
+          getAssetUrl: (assetId: string) => {
+            throw new Error(
+              `IndexedDB is required by the default asset manager, but it's not supported by this browser. To use assets, please provide your own asset manager to the project config.`,
+            )
+          },
+          createAsset: (asset: Blob) => {
+            throw new Error(
+              `IndexedDB is required by the default asset manager, but it's not supported by this browser. To use assets, please provide your own asset manager to the project config.`,
+            )
+          },
+        }
+      }
+
+      // get all possible asset ids referenced by either static props or keyframes
+      const possibleAssetIDs = getAllPossibleAssetIDs(project)
+
+      // Clean up assets not referenced by the project. We can only do this at the start because otherwise
+      // we'd break undo/redo.
+      const idbKeys = await idb.keys<string>()
```

Comment:
Also, let's use a studio-specific key for idb

Reply:
Yes